### PR TITLE
Add support for specifying both foreground and background colors.

### PR DESCRIPTION
**Pull request by cswank**
for example:

    \033[48;5;200;38;5;100mHello

results in a foreground color (100) and a background color (200)

Since this also introduces the first test, what would you think about using testify/assert in the tests?

### CODE REVIEW

File: escape_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package gocui
```
Comment (mjarkk) [inline escape_test.go]: I think we don't want any tests in this library at this point in time.  
Maybe in the further but for now this file is not needed an can be removed.

**Review by mjarkk — CHANGES_REQUESTED** [review]
Agree with the changes as long as this is supported in all major terminals,  
Also left a few comments.

File: .gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
 *.swp
+*~
```
Comment (mjarkk) [inline .gitignore]: Why add this to the .gitignore?

Comment (cswank) [reply .gitignore]: This is for emacs temp files.  They get created every time it modifies a file.

File: escape_test.go (hunk 1 above)
Comment (cswank) [reply escape_test.go]: ok, done

**Comment by cswank** [comment]
Yes, I'm pretty sure that specifying foreground and background at the same time is supported on major terminals.  I tested a few with


printf '\033[48;5;130;38;5;40mHi\033[0m\n'

File: .gitignore (hunk 2 above)
Comment (cswank) [reply .gitignore]: I removed it and added it to my global gitignore (which I should have done in the first place).

**Review by mjarkk — APPROVED** [review]
LGTM,  
ping @glvr182 @skanehira to review the changes

File: escape.go [hunk 3]
```diff
@@ -191,39 +191,59 @@ func (ei *escapeInterpreter) output256() error {
...
 
-			switch {
-			case p == 1:
-				ei.curFgColor |= AttrBold
-			case p == 4:
-				ei.curFgColor |= AttrUnderline
-			case p == 7:
-				ei.curFgColor |= AttrReverse
-			}
+func (ei *escapeInterpreter) splitFgBg() [][]string {
```
Comment (glvr182) [inline escape.go]: does this function need to be a method of `ei`, cant this just be a function we can call like 
```
func splitFgBg() [][]string { ... return out}
```

File: escape.go [hunk 4]
```diff
@@ -191,39 +191,59 @@ func (ei *escapeInterpreter) output256() error {
...
+		if err != nil {
+			return errCSIParseError
+		}
 
-		for _, param := range ei.csiParam[3:] {
-			p, err := strconv.Atoi(param)
-			if err != nil {
-				return errCSIParseError
+		switch fgbg {
+		case 38:
```
Comment (glvr182) [inline escape.go]: can we change this magic number stuff while we are at it

**Review by glvr182 — CHANGES_REQUESTED** [review]
Looks good, added some comments

File: escape.go (hunk 3 above)
Comment (cswank) [reply escape.go]: spltFfBg is very specific to the escapeInterpreter, so to me it makes sense to keep it as is.

File: escape.go (hunk 4 above)
Comment (cswank) [reply escape.go]: I pushed a change that I think is what you want.

**Comment by cswank** [comment]
This PR should not be merged yet.

I just thought of an edge case and recovered escape_test.go that you asked me to remove and added another test case in which the background color being set is 38.

`"\033[48;5;38;38;5;100mHi!!"`

38 is a valid color but also the code for setting the foreground color.  It causes escapeInterpreter to panic with my changes.   

Would you prefer I close this PR and open another one or just ping you when I think I have this edge case fixed?

**Comment by cswank** [comment]
I think that latest commit fixes the edge cases I thought of.

**Comment by cswank** [comment]
I also realized that this only supports 8 bit colors (ei.output256()).  I think outputNormal() needs the same behavior.  

I'll work on that now.

**Comment by cswank** [comment]
It looks like OutputNormal already supports specifying foreground and background at the same time.  So this should be good to go.

File: escape.go (hunk 3 above)
Comment (glvr182) [reply escape.go]: I get that, but it does not rely on ei, it just relies on data. It is not driven by the struct

Comment (cswank) [reply escape.go]: Not a big deal to me.  I'll change it.  Just a sec.

Comment (cswank) [reply escape.go]: done

**Comment by mjarkk** [comment]
Thanks for making the PR
